### PR TITLE
[Merged by Bors] - feat(Data/Sum/Basic): create direct versions of get(Left|Right)? analogous to Option.get

### PR DESCRIPTION
**Pull request by wrenna-robson**
---

- [x] #6665

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Sum/Basic.lean [hunk 1]
```diff
@@ -16,12 +16,14 @@ This file proves basic results about the sum type `α ⊕ β`.
...
 ## Main declarations
 
-* `Sum.getLeft`: Retrieves the left content of `x : α ⊕ β` or returns `none` if it's coming from
-  the right.
-* `Sum.getRight`: Retrieves the right content of `x : α ⊕ β` or returns `none` if it's coming from
-  the left.
 * `Sum.isLeft`: Returns whether `x : α ⊕ β` comes from the left component or not.
 * `Sum.isRight`: Returns whether `x : α ⊕ β` comes from the right component or not.
+* `Sum.getLeft`: Retrieves the left content of a `x : α ⊕ β` that is known to come from the left.
+* `Sum.getRight?`: Retrieves the right content of `x : α ⊕ β` that is known to come from the right.
```
Comment (eric-wieser) [inline Mathlib/Data/Sum/Basic.lean:22]:
```suggestion
* `Sum.getRight`: Retrieves the right content of `x : α ⊕ β` that is known to come from the right.
```

File: Mathlib/Data/Sum/Basic.lean [hunk 2]
```diff
@@ -91,33 +81,85 @@ def isRight : Sum α β → Bool
...
 @[simp] theorem isLeft_inl (x : α) : (inl x : α ⊕ β).isLeft = true := rfl
 @[simp] theorem isLeft_inr (x : β) : (inr x : α ⊕ β).isLeft = false := rfl
 @[simp] theorem isRight_inl (x : α) : (inl x : α ⊕ β).isRight = false := rfl
 @[simp] theorem isRight_inr (x : β) : (inr x : α ⊕ β).isRight = true := rfl
 
-@[simp] theorem getLeft_eq_none_iff : x.getLeft = none ↔ x.isRight := by
-  cases x <;> simp only [getLeft, isRight, eq_self_iff_true]
-#align sum.get_left_eq_none_iff Sum.getLeft_eq_none_iff
+@[simp] lemma getLeft_inl (x : α) (h : (inl x : α ⊕ β).isLeft) : (inl x).getLeft h = x := rfl
+@[simp] lemma getRight_inr (x : β) (h : (inr x : α ⊕ β).isRight) : (inr x).getRight h = x := rfl
```
Comment (eric-wieser) [inline Mathlib/Data/Sum/Basic.lean:112]: Can you make these all `theorem` for consistency?

File: Mathlib/Data/Sum/Basic.lean (hunk 1 above)
Comment (linesthatinterlace) [reply Mathlib/Data/Sum/Basic.lean]: Done.

File: Mathlib/Data/Sum/Basic.lean (hunk 2 above)
Comment (linesthatinterlace) [reply Mathlib/Data/Sum/Basic.lean]: Done.

**Comment by eric-wieser** [comment]
Would you mind splitting out a PR that adds the question marks and nothing else? It will make the diff clearer.

**Comment by linesthatinterlace** [comment]
Oh, certainly. I assume I should keep this one up and just rebase it. I did my commits like that so it should actually be the first commit of this branch.

**Comment by linesthatinterlace** [comment]
I thought you might mean that but I wasn't sure and I thought better to do that first.

**Comment by linesthatinterlace** [comment]
See #6665.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**